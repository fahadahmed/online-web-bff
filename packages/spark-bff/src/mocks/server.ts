import { setupServer } from 'msw/node';
import a2pCustomerHandlers from 'datasources/a2p/customer/mocks/handlers';
import accessHandlers from 'datasources/user/access/mocks/handlers';
import accountSummaryHandlers from 'datasources/customer/accountsService/mocks/handlers';
import addressHandlers from 'datasources/location/address/mocks/handlers';
import airpointsHandlers from 'datasources/airpoints/mocks/handlers';
import availableNumbers from 'datasources/shop/availableNumbers/mocks/handlers';
import balanceServiceHandlers from 'datasources/finance/balanceService/mocks/handlers';
import billPaymentHandlers from 'datasources/finance/paymentService/billPayment/mocks/handlers';
import cartServicesHandlers from 'datasources/shop/cart/mocks/handlers';
import checkoutServicesHandlers from 'datasources/shop/checkout/mocks/handlers';
import connectionDiagnosticsHandlers from 'datasources/line/assure/connectionDiagnostics/mocks/handlers';
import connectionPromiseHandler from 'datasources/line/assure/connectionPromise/mocks/handlers';
import contentServiceHandlers from 'datasources/utility/content/mocks/handlers';
import digitalBillHandlers from 'datasources/finance/digitalBill/mocks/handlers';
import identityServiceHandlers from 'datasources/user/identityService/mocks/handlers';
import lineConfiguarationHandlers from 'datasources/line/lineConfiguration/mocks/handlers';
import lineDetailServiceHandlers from 'datasources/line/lineDetails/mocks/handlers';
import lineDetailsHandlers from 'datasources/line/lineSummary/mocks/handlers';
import lineSpendHistoryHandlers from 'datasources/line/spend/mocks/handlers';
import lineSubscriptionsService from 'datasources/line/lineSubscriptionsService/mocks/handlers';
import lineUsageDetailsMocks from 'datasources/line/usage/usageDetailsService/mocks';
import lineUsageHistoryMocks from 'datasources/line/usage/usageHistoryService/mocks';
import lineUsageMocks from 'datasources/line/usage/mocks/handlers';
import lineUsageTransactionMocks from 'datasources/line/usage/usageTransactionsService/mocks';
import networkSettingsHandlers from 'datasources/technology/networkSettingsService/mocks/handlers';
import orderHandlers from 'datasources/shop/order/mocks/handlers';
import paymentServiceHandlers from 'datasources/finance/paymentService/mocks/handlers';
import pointOfAddressHandlers from 'datasources/location/pointOfInterest/mocks/handlers';
import pointOfOutageHandlers from 'datasources/location/pointOfInterest/outagesPointOfInterest/mocks/handlers';
import preferenceHandlers from 'datasources/user/preference/mocks/handlers';
import prepaidBalanceServiceHandlers from 'datasources/finance/prepaidBalanceService/mocks/handlers';
import productDetailServiceHandlers from 'datasources/product/details/mocks/handlers';
import productFiltersServiceHandlers from 'datasources/product/filters/mocks/handlers';
import productOffersHandlers from 'datasources/product/offers/mocks/handlers';
import productPromotionsMPDServiceHandlers from 'datasources/product/promotions/mpd/mocks/handlers';
import profileServiceHandlers from 'datasources/marketing/profileService/mocks/handlers';
import resourceHandlers from 'datasources/utility/resource/mocks/handlers';
import roamingRatesHandlers from 'datasources/product/roamingRates/mocks/handlers';
import subscriptionActivationHandlers from 'datasources/line/subscriptionActivationService/mocks/handlers';
import topupHandlers from 'datasources/finance/paymentService/topup/mocks/handlers';
import transactionsListHandlers from 'datasources/line/transactionsService/mocks/handlers';
import troubleshootHandlers from 'datasources/line/assure/connectionTroubleshooter/mocks/handlers';
import userInfoHandlers from 'datasources/user/userInfo/mocks/handlers';
import walletServiceHandlers from 'datasources/finance/walletService/mocks/handlers';

const handlers = [
  ...a2pCustomerHandlers,
  ...accessHandlers,
  ...accountSummaryHandlers,
  ...addressHandlers,
  ...availableNumbers,
  ...airpointsHandlers,
  ...balanceServiceHandlers,
  ...billPaymentHandlers,
  ...cartServicesHandlers,
  ...checkoutServicesHandlers,
  ...connectionDiagnosticsHandlers,
  ...connectionPromiseHandler,
  ...contentServiceHandlers,
  ...digitalBillHandlers,
  ...identityServiceHandlers,
  ...lineConfiguarationHandlers,
  ...lineDetailServiceHandlers,
  ...lineDetailsHandlers,
  ...lineSpendHistoryHandlers,
  ...lineSubscriptionsService,
  ...lineUsageDetailsMocks,
  ...lineUsageHistoryMocks,
  ...lineUsageMocks,
  ...lineUsageTransactionMocks,
  ...networkSettingsHandlers,
  ...orderHandlers,
  ...paymentServiceHandlers,
  ...pointOfAddressHandlers,
  ...pointOfOutageHandlers,
  ...preferenceHandlers,
  ...prepaidBalanceServiceHandlers,
  ...productDetailServiceHandlers,
  ...productFiltersServiceHandlers,
  ...productOffersHandlers,
  ...productPromotionsMPDServiceHandlers,
  ...profileServiceHandlers,
  ...resourceHandlers,
  ...roamingRatesHandlers,
  ...subscriptionActivationHandlers,
  ...topupHandlers,
  ...transactionsListHandlers,
  ...troubleshootHandlers,
  ...userInfoHandlers,
  ...walletServiceHandlers,
];

const server = setupServer(...handlers);
export default server;
