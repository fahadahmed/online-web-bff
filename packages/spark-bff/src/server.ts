import { ApolloServer } from 'apollo-server-fastify';
import {
  ApolloServerPluginLandingPageDisabled,
  ApolloServerPluginLandingPageGraphQLPlayground,
} from 'apollo-server-core';
import { logger } from 'apolloLogger';
import context from 'context';
import dataSources from 'datasources';
import { ApolloLoggerPlugin, ApplicationLogEvent } from 'logging';
import schema from 'schema';
import { formatResponse } from 'utils/formatResponse';
import { isDevelopmentEnvironment } from 'utils/isDevelopmentEnvironment';

process.on('uncaughtException', (err) => {
  logger.error(new ApplicationLogEvent({ message: err.message, context: {} }));
  throw err;
});

const apolloServer = new ApolloServer({
  schema,
  dataSources,
  formatResponse,
  logger,
  context,
  introspection: isDevelopmentEnvironment(),
  plugins: [
    new ApolloLoggerPlugin(
      `BFF Server Started. BUILD_ID: ${process.env.BUILD_BUILDID}`,
    ),
    isDevelopmentEnvironment()
      ? ApolloServerPluginLandingPageGraphQLPlayground()
      : ApolloServerPluginLandingPageDisabled(),
  ],
});

export default apolloServer;
