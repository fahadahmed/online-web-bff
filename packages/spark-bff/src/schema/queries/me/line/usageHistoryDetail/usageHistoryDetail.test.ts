import { createTestClient, gql } from 'test/apolloTestUtils';
import { server } from 'test/server';
import usageHistoryMocks from 'datasources/line/usage/usageHistoryService/mocks';

beforeEach(() => {
  server.use(...usageHistoryMocks);
});

test('plan data usage history detail', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          line(lineNumber: "033227715") {
            usageHistoryDetail(
              usageType: DATA
              interval: MONTHLY
              periodBreakdownId: "7777777"
            ) {
              interval
              breakdownType
              startDateTime
              endDateTime
              unbilled
              periodUsage {
                value
                unit
              }
              periodGroupUsage {
                unit
                value
              }
              contributingProducts {
                offerId
                productName
                acquisitionDate
                periodUsage {
                  value
                  unit
                }
                periodGroupUsage {
                  unit
                  value
                }
                periodLimit {
                  value
                  unit
                }
                periodGroupLimit {
                  unit
                  value
                }
                periodRemaining {
                  value
                  unit
                }
                periodGroupRemaining {
                  unit
                  value
                }
              }
            }
          }
        }
      }
    `,
  });
  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});

test('additional paid voice usage history detail', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          line(lineNumber: "033227715") {
            usageHistoryDetail(
              usageType: VOICE
              interval: MONTHLY
              periodBreakdownId: "aaaaa"
            ) {
              interval
              breakdownType
              startDateTime
              endDateTime
              periodUsage {
                value
                unit
              }
              periodGroupUsage {
                unit
                value
              }
              contributingProducts {
                offerId
                productName
                acquisitionDate
                periodUsage {
                  value
                  unit
                }
                periodGroupUsage {
                  unit
                  value
                }
                periodLimit {
                  value
                  unit
                }
                periodGroupLimit {
                  unit
                  value
                }
                periodRemaining {
                  value
                  unit
                }
                periodGroupRemaining {
                  unit
                  value
                }
              }
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});

test('additional included sms usage history detail', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          line(lineNumber: "033227715") {
            usageHistoryDetail(
              usageType: SMS
              interval: MONTHLY
              periodBreakdownId: "1234567"
            ) {
              interval
              breakdownType
              startDateTime
              endDateTime
              periodUsage {
                value
                unit
              }
              contributingProducts {
                offerId
                productName
                acquisitionDate
                periodUsage {
                  value
                  unit
                }
                periodLimit {
                  value
                  unit
                }
                periodRemaining {
                  value
                  unit
                }
              }
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});
