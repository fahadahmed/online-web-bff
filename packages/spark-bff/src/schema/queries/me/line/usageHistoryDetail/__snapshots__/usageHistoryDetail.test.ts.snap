// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`additional included sms usage history detail 1`] = `
Object {
  "usageHistoryDetail": Object {
    "breakdownType": "ADDITIONAL_INCLUDED",
    "contributingProducts": Array [
      Object {
        "acquisitionDate": "2021-08-08T12:00:00Z",
        "offerId": "mbundle050820",
        "periodLimit": Object {
          "unit": "TEXT",
          "value": 500,
        },
        "periodRemaining": Object {
          "unit": "TEXT",
          "value": 100,
        },
        "periodUsage": Object {
          "unit": "TEXT",
          "value": 400,
        },
        "productName": "$59.99 Endless Mobile Plan",
      },
    ],
    "endDateTime": "2021-09-08T11:59:59Z",
    "interval": "MONTHLY",
    "periodUsage": Object {
      "unit": "TEXT",
      "value": 400,
    },
    "startDateTime": "2021-08-08T12:00:00Z",
  },
}
`;

exports[`additional paid voice usage history detail 1`] = `
Object {
  "usageHistoryDetail": Object {
    "breakdownType": "ADDITIONAL_PAID",
    "contributingProducts": Array [
      Object {
        "acquisitionDate": "2021-07-08T12:00:00Z",
        "offerId": "mbundle050820",
        "periodGroupLimit": null,
        "periodGroupRemaining": null,
        "periodGroupUsage": null,
        "periodLimit": null,
        "periodRemaining": null,
        "periodUsage": Object {
          "unit": "MIN",
          "value": 200,
        },
        "productName": "$59.99 Endless Mobile Plan",
      },
    ],
    "endDateTime": "2021-08-08T11:59:59Z",
    "interval": "MONTHLY",
    "periodGroupUsage": Object {
      "unit": "MIN",
      "value": 0,
    },
    "periodUsage": Object {
      "unit": "MIN",
      "value": 200,
    },
    "startDateTime": "2021-07-08T12:00:00Z",
  },
}
`;

exports[`plan data usage history detail 1`] = `
Object {
  "usageHistoryDetail": Object {
    "breakdownType": "PLAN",
    "contributingProducts": Array [
      Object {
        "acquisitionDate": "2021-08-08T12:00:00Z",
        "offerId": "mbundle050820",
        "periodGroupLimit": null,
        "periodGroupRemaining": null,
        "periodGroupUsage": null,
        "periodLimit": Object {
          "unit": "MB",
          "value": 12288,
        },
        "periodRemaining": Object {
          "unit": "MB",
          "value": 2047.960936,
        },
        "periodUsage": Object {
          "unit": "MB",
          "value": 10240.039064,
        },
        "productName": "$59.99 Endless Mobile Plan",
      },
      Object {
        "acquisitionDate": "2021-08-08T12:00:00Z",
        "offerId": "mbundle050821",
        "periodGroupLimit": null,
        "periodGroupRemaining": null,
        "periodGroupUsage": null,
        "periodLimit": Object {
          "unit": "MB",
          "value": 12288,
        },
        "periodRemaining": Object {
          "unit": "MB",
          "value": 2047.960936,
        },
        "periodUsage": Object {
          "unit": "MB",
          "value": 10240.039064,
        },
        "productName": "$59.99 Endless Mobile Plank",
      },
    ],
    "endDateTime": "2021-09-08T11:59:59Z",
    "interval": "MONTHLY",
    "periodGroupUsage": Object {
      "unit": "MB",
      "value": 0,
    },
    "periodUsage": Object {
      "unit": "MB",
      "value": 10240.039064,
    },
    "startDateTime": "2021-08-08T12:00:00Z",
    "unbilled": false,
  },
}
`;
